src/main/java/ca/corefacility/bioinformatics/irida/service/impl/RemoteAPITokenServiceImpl.java

```diff
@@ -99,83 +100,90 @@ public RemoteAPIToken updateTokenFromRefreshToken(RemoteAPI api) {
 		try {
 			token = getToken(api);
 
-			String refreshToken = token.getRefreshToken();
+			String refreshTokenValue = token.getRefreshToken();
+
+			if (refreshTokenValue != null) {
+				RefreshToken refreshToken = new RefreshToken(refreshTokenValue);
 
-			if (refreshToken != null) {
 				URI serviceTokenLocation = UriBuilder.fromUri(api.getServiceURI()).path("oauth").path("token").build();
 
-				OAuthClientRequest tokenRequest = OAuthClientRequest.tokenLocation(serviceTokenLocation.toString())
-						.setClientId(api.getClientId())
-						.setClientSecret(api.getClientSecret())
-						.setRefreshToken(refreshToken)
-						.setGrantType(GrantType.REFRESH_TOKEN)
-						.buildBodyMessage();
+				ClientAuthentication clientAuth = new ClientSecretBasic(new ClientID(api.getClientId()),
+						new Secret(api.getClientSecret()));
+
+				TokenRequest tokenRequest = new TokenRequest(serviceTokenLocation, clientAuth,
+						new RefreshTokenGrant(refreshToken));
+
+				TokenResponse tokenResponse = TokenResponse.parse(tokenRequest.toHTTPRequest().send());
+
+				if (!tokenResponse.indicatesSuccess()) {
+					// We got an error response...
+					TokenErrorResponse errorResponse = tokenResponse.toErrorResponse();
+					logger.error("Updating token by refresh token failed", errorResponse.getErrorObject().toString());
```

Comment:
The token in the database will not be updated and when the project attempts to sync it will fail with an Authentication error. Basically that the current token is invalid. This matches the existing behaviour.

Reply:
Okay, sounds good.